docs/standard/native-interop/pinvoke-source-generation.md

```diff
@@ -0,0 +1,72 @@
+---
+title: P/Invoke source generation
+description: Learn about compile-time source generation for platform invokes in .NET.
+ms.date: 07/25/2022
+---
+
+# Source generation for platform invokes
+
+.NET 7 introduces a [source generator](../../csharp/roslyn-sdk/source-generators-overview.md) for P/Invokes which recognizes the <xref:System.Runtime.InteropServices.LibraryImportAttribute>.
+
+When a P/Invoke is defined and called like below, the built-in interop system in the .NET runtime generates an IL stub&mdash;a stream of IL instructions that is JIT-ed&mdash; at run time to facilitate the transition from managed to unmanaged.
+
+```csharp
+[DllImport("nativelib", EntryPoint = "to_lower", CharSet = CharSet.Unicode)]
+internal static extern string ToLower(string str);
+
+string lower = ToLower("StringToConvert");
```

Comment:
```suggestion
[DllImport(
    "nativelib",
    EntryPoint = "to_lower",
    CharSet = CharSet.Unicode)]
internal static extern string ToLower(string str);

// string lower = ToLower("StringToConvert");
```
Formatted to avoid horizontal scrollbar, and comment out example code that would otherwise not compile.